### PR TITLE
Final split between legacy compose and BAN-ID data processing 

### PR DESCRIPTION
**Pull request by antoineludeau**
WARNING : pre-requisite on this other PR : https://github.com/BaseAdresseNationale/ban-plateforme/pull/383

# Context

For now, data coming from the new API system (REST API + Postgres + export to Mongo) was going into temporary collections to be able to tests the system with real case scenario.
Now that tests have been done, we need to make the final switch to connect the new system to the current mongo exploited collections.

# Enhancement 

With this PR, there are the following changes : 
- a new route /api/legacy-compose/:codeCommune/ used only by "id-fix". The current route /ban/communes/:codeCommune/compose is now sending a request to "if-fix" ONLY.
- the bal garbage collector has been modified so that it sends the de-synchronized district to "id-fix" instead of asking for a legacy compose.
- a new property "withBanId" has been added to "communes" collection data to be able to differentiate data that comes from the legacy (withBanId=false) and data that comes from the new system (withBanId=true). This property is also used in the legacy compose to not compose a district that would have passed by the new system.
- a new flag "ignoreIdConfig" used in legacy compose to be able to revert a district from the new system to the legacy system (in a case where a district have published its district with BAN-ID and then would like to publish it without them).

### CODE REVIEW

File: lib/api/legacy-routes.cjs [hunk 1]
```diff
@@ -246,15 +246,14 @@ app.get(
 app.post(
   '/ban/communes/:codeCommune/compose',
   ensureIsAdmin,
-  analyticsMiddleware.composeCommune,
```
Comment (nkokla) [inline lib/api/legacy-routes.cjs:249]: Il faudrait qu'on conserve une trace d'appel à cette API.

File: lib/api/consumers/export-to-exploitation-db-consumer.js [hunk 2]
```diff
@@ -36,16 +36,9 @@ const ADDRESS_BBOX_BUFFER = 50
...
-    address: 'numeros_temp'
-  },
-  banID: {
-    district: 'district',
-    commonToponym: 'common_toponym',
-    address: 'address'
-  }
+  district: 'communes',
+  commonToponym: 'voies',
+  address: 'numeros'
```
Comment (nkokla) [inline lib/api/consumers/export-to-exploitation-db-consumer.js:41]: Il faudra qu'on prenne le temps de voir si on peut faire correspondre la base mongo au nouveau modele de donnée (avec les objets district, common-toponym et address).

**Review by nkokla — APPROVED** [review]

File: lib/api/legacy-routes.cjs (hunk 1 above)
Comment (antoineludeau) [reply lib/api/legacy-routes.cjs:249]: Le ping analytique se trouve dans la nouvelle route (/api/legacy-compose/:codeCommune/) qui est appelée par id-fix.

File: lib/api/consumers/export-to-exploitation-db-consumer.js (hunk 2 above)
Comment (antoineludeau) [reply lib/api/consumers/export-to-exploitation-db-consumer.js:41]: Dans quel scénario ?

File: lib/compose/bal-garbage-collector/index.js [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
-  const idRevisionBAN = Object.fromEntries(revisionBAN.map(({idRevision, codeCommune}) => [codeCommune, idRevision]))
+    const idRevisionBAL = revisionBAL.map(({_id, codeCommune}) => [codeCommune, _id])
+    const idRevisionBAN = Object.fromEntries(revisionBAN.map(({idRevision, codeCommune}) => [codeCommune, idRevision]))
 
-  const codesCommunes = idRevisionBAL.filter(([codeCommune, idRevision]) => idRevisionBAN[codeCommune] !== idRevision)
-
-  const communeComposed = await Promise.all(codesCommunes.map(([codeCommune]) => askComposition(codeCommune)))
-  return communeComposed
+    const codesCommunes = idRevisionBAL.filter(([codeCommune, idRevision]) => idRevisionBAN[codeCommune] !== idRevision)
+    const communeComposed = await Promise.all(codesCommunes.map(([codeCommune]) => {
```
[the remark this answers is not in the document]
Comment (antoineludeau) [reply lib/compose/bal-garbage-collector/index.js:17]: L'auto spam/ban ?

File: lib/api/legacy-routes.cjs [hunk 4]
```diff
@@ -290,6 +288,35 @@ app.post(
   })
 )
 
+app.post(
+  '/api/legacy-compose/:codeCommune/',
+  ensureIsAdmin,
+  analyticsMiddleware.composeCommune,
+  w(async (req, res) => {
+    const {codeCommune} = req.commune
```
[the remark this answers is not in the document]
Comment (antoineludeau) [reply lib/api/legacy-routes.cjs:296]: Oui les communes doivent déjà exister dans la BAN (ce comportement ne change pas)

File: lib/compose/bal-garbage-collector/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (antoineludeau) [reply lib/compose/bal-garbage-collector/index.js]: J'ai ajouté, coté id-fix, une route /compute-from-cogs qui permet de recevoir une liste de cogs et de faire un processing synchrone.

**Review by nkokla — APPROVED** [review]

**Pull request merged**